src/core/include/iDynTree/Core/Transform.h

```diff
@@ -102,11 +102,25 @@ namespace iDynTree
         Transform(const Rotation & _rot, const Position & origin);
 
         /**
+        * Constructor from a raw row-major ordered buffer of doubles
+        * @param in_data the raw buffer of 16 doubles
+        * @param in_size size of the buffer (must be 16)
+        */
+        Transform(double* in_data, const unsigned int in_size);
+
+        /**
          * Copy constructor: create a Transform from another Transform.
          */
         Transform(const Transform & other);
 
         /**
+         * Set rotation and translation from a row-major ordered buffer containing the transform
+         * @param in_data the raw buffer of 16 doubles
+         * @param in_size size of the buffer (must be 16)
+         */
+        Transform fromHomogeneousTransform(double* in_data, const unsigned int in_size);
```

Comment:
We have a few `from***` methods ( [`fromQuaternion`](https://github.com/robotology/idyntree/blob/9b80ac5bc6f8244b941ae9bb3bf686ac48226db7/src/core/include/iDynTree/Core/Rotation.h#L152) ,  [`fromMatlab`](https://github.com/robotology/idyntree/blob/89c27bbea7e9848d92ea2cd6142cb6da2011e991/bindings/matlab/matlab_matvec.i#L88) ) that actually modify the object on which they are called rather then constructing a new one, so it could be more consistent to follow this convention. 
Even if you change this in an actual method, for the future if in a method you are not using `this` at all consider transforming it in a `static` method, so it can be used also without creating an instance of the class.

Reply:
Just read https://github.com/robotology/idyntree/pull/363#discussion_r136073112 . : )